crates/sargon/src/wrapped_radix_engine_toolkit/low_level/manifest_summary/account_deposit/simple_fungible_resource_bounds.rs

```diff
@@ -0,0 +1,168 @@
+use crate::prelude::*;
+
+/// Represents the bounds for a simple fungible resource, which can
+/// be exact, at most, at least, between, or unknown amount.
+#[derive(Clone, Debug, PartialEq, Eq, uniffi::Enum)]
+pub enum SimpleFungibleResourceBounds {
+    Exact {
+        amount: Decimal,
+    },
+    AtMost {
+        amount: Decimal,
+    },
+    AtLeast {
+        amount: Decimal,
+    },
+    Between {
+        min_amount: Decimal,
+        max_amount: Decimal,
+    },
+    UnknownAmount,
+}
+
+impl SimpleFungibleResourceBounds {
+    pub fn exact(amount: Decimal) -> Self {
+        Self::Exact { amount }
+    }
+
+    pub fn at_most(amount: Decimal) -> Self {
+        Self::AtMost { amount }
+    }
+
+    pub fn at_least(amount: Decimal) -> Self {
+        Self::AtLeast { amount }
+    }
+
+    pub fn between(min_amount: Decimal, max_amount: Decimal) -> Self {
+        Self::Between {
+            min_amount,
+            max_amount,
+        }
+    }
+
+    pub fn unknown_amount() -> Self {
+        Self::UnknownAmount
+    }
+}
+
+impl From<ScryptoSimpleFungibleResourceBounds>
+    for SimpleFungibleResourceBounds
+{
+    fn from(value: ScryptoSimpleFungibleResourceBounds) -> Self {
+        match value {
+            ScryptoSimpleFungibleResourceBounds::Exact(amount) => Self::Exact {
+                amount: amount.into(),
+            },
+            ScryptoSimpleFungibleResourceBounds::AtMost(amount) => {
+                Self::AtMost {
+                    amount: amount.into(),
+                }
+            }
+            ScryptoSimpleFungibleResourceBounds::AtLeast(amount) => {
+                Self::AtLeast {
+                    amount: amount.into(),
+                }
+            }
+            ScryptoSimpleFungibleResourceBounds::Between(
+                min_amount,
+                max_amount,
+            ) => Self::Between {
+                min_amount: min_amount.into(),
+                max_amount: max_amount.into(),
+            },
+            ScryptoSimpleFungibleResourceBounds::UnknownAmount => {
+                Self::UnknownAmount
+            }
+        }
+    }
+}
+
+impl HasSampleValues for SimpleFungibleResourceBounds {
+    fn sample() -> Self {
+        Self::sample_exact()
+    }
+
+    fn sample_other() -> Self {
+        Self::sample_unknown_amount()
+    }
+}
+
+impl SimpleFungibleResourceBounds {
+    fn sample_exact() -> Self {
+        Self::exact(Decimal::from(1337))
```

Comment:
and if you use `impl Into<Decimal>` advice on constructors above, you can type just `1337` here :)